tools/IceRpc.SliceBuildTelemetry.Reporter/Program.cs

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) ZeroC, Inc.
+
+using IceRpc;
+using IceRpc.BuildTelemetry;
+using System.CommandLine;
+using System.Net.Security;
+using System.Reflection;
+using System.Runtime.InteropServices;
+
+var rootCommand = new RootCommand("Slice build telemetry reporter");
+
+var hashOption = new Option<string>(
+    name: "--hash",
+    description: "The compilation hash.")
+{
+    IsRequired = true,
+};
+rootCommand.AddOption(hashOption);
+
+var containsSlice1Option = new Argument<bool>(
+    name: "--contains-slice1",
+    description: "Whether or not the build contains Slice1 definitions");
+rootCommand.AddArgument(containsSlice1Option);
+
+var containsSlice2Option = new Argument<bool>(
+    name: "--contains-slice2",
+    description: "Whether or not the build contains Slice2 definitions");
+rootCommand.AddArgument(containsSlice1Option);
+
+var sourceFileCountOption = new Option<int>(
+    name: "--src-file-count",
+    description: "The number of source files included in the compilation.");
+rootCommand.AddOption(sourceFileCountOption);
+
+var referenceFileCountOption = new Option<int>(
+    name: "--ref-file-count",
+    description: "The number of reference files included in the compilation.");
+rootCommand.AddOption(referenceFileCountOption);
+
+rootCommand.SetHandler(
+    async (string hash, bool containsSlice1, bool containsSlice2, int sourceFileCount, int referenceFileCount) =>
+    {
+        const string uri = "icerpc://telemetry.icerpc.dev";
+        using var cts = new CancellationTokenSource(TimeSpan.FromSeconds(3));
+        await using var connection = new ClientConnection(new Uri(uri), new SslClientAuthenticationOptions());
+
+        // Create a reporter proxy with this client connection.
+        var reporter = new ReporterProxy(connection);
+
+        // Determine the IceRPC version using the assembly version.
+        var assembly = Assembly.GetAssembly(typeof(SliceTelemetryData));
+        string version = assembly!.GetName().Version!.ToString();
+
+        var sliceTelemetryData = new SliceTelemetryData(
+            version,
+            RuntimeInformation.OSDescription,
```

Comment:
`OSDescription` is more verbose than the previous `SystemHelpers.GetOperatingSystem()`. I think it makes more sense to normalize this data in the server.